---
title: "MATH1324 Assignment 4"
author: "Alistair Grevis-James s3644119"
output:
  html_notebook: default
  html_document: default
  pdf_document: default
---
<style>
body {
text-align: justify}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r defining markup language for qwraps2, echo = FALSE}
options(qwraps2_markup = "markdown")
```

```{r importing the libraries, echo=FALSE}
library(rmarkdown)
library(GGally)
library(car)
library(ggplot2)
library(dplyr)
library(gsheet)
library(gridExtra)
library("htmltools")
library(reshape2)
library(granova)
library(psychometric)
library(Hmisc)
library(qwraps2)
library(outliers)
```

# Statistical Evaluation of Mice Protein Expression Data Set

### Executive Summary

### Hypothesis
Comment on not needed to be normalized in preprocessing

### Introduction
The chosen dataset ‘Mice Protein Expression Data Set’ was generated from experiments by Higuera et al[^1] and Ahmed et al[^2]. These projects aimed to understand the impacts of Down Syndrome on learning through analysis of protein expression in mice. Down Syndrome (DS) has a prevalence globally of 1 in a 1000 live human births, and is the most common genetically defined cause of intellectual disabilities^1,^[^3]. DS in humans is caused by the presence of an additional chromosome 21, referred to as trisomy^3^. Protein expression is disrupted by human trisomy 21, leading to the physical and intellectual manifestations associated with DS. Due to its prevalence and health implications, a strong imperative exists to further understand and treat the condition.  

Davisson et al., successfully manipulated a mouse genome to produce several models of DS in rodents[^4]. The Ts65Dn mouse is the best-characterized of the DS rodent models[^5]. In generating the 'Mice Protein Expression Data Set' Higuera et al employed Ts65Dn and normal mice in experimental and control groups, exposing them to a range of variables. The rodents were then euthanized and their cortex protein levels were analysed in a quantitative fashion. From three binary variables, eight classes of mice were used to generate the data set. The eight classes are summarised in Figure 1.  

The work of Higuera et al and Ahmed et al aimed to assess the efficacy of pharmacotherapies for treatment of DS - an avenue of treatment that has never been successfully implemented. Ahmed et al statistically evaluated the ‘Mice Protein Expression Data Set’ through the use of the Wilcoxon test, with the subsequent production of 'self-organising maps' to assess proteins critical to learning. Higuera et al also used pair-wise comparisons to evaluate the significance of difference between pairs of the classes, specifically to assess protein dynamics.   
<br>
<center>
![](http://journals.plos.org/plosone/article/figure/image?size=medium&id=10.1371/journal.pone.0129126.g001)
</center>  
<br>
_Fig 1. Classes of mice. (A) There are eight classes of mice based on genotype (control, c, and trisomy, t), stimulation to learn (Context-Shock, CS, and Shock-Context, SC) and treatment (saline, s, and memantine, m). Learning outcome indicates the response to learning for each class. (B) Number of mice in each class. (C) Format of protein expression data: rows are individual mice, and columns, P 1 … P77, are the measured levels of the 77 proteins[^6]._

### RStudio Libraries  
The libraries `rmarkdown`, `GGally`, `car`, `ggplot2`, `dplyr`, `gsheet`, `gridExtra`,`htmltools`, `reshape2`, `granova`, `psychometric`, `Hmisc`, `qwraps` & `outliers` were employed during the analysis.

### Data Set  
The Excel file ‘Data_Cortex_Nuclear.xls’ was downloaded from UCI’s machine learning repository[^7] into Google Drive (s3644119 at RMIT University) and the imported into the RStudio interactive environment as “ds”.

```{r importing the dataset, tidy=TRUE, echo=TRUE}
url <- 'https://docs.google.com/spreadsheets/d/158zPd4XCYoaXNzHA2OJPkOmzSWYjwFoEuOiMzV1O-rU/edit?usp=sharing'
ds <- gsheet2tbl(url)
```

### Data Checking
The initial data checking is summarized below.

 Function | Purpose / Outcome
------------- | -------------------------------------------------
`summary(ds)` | Confirming the successful importation of the data set. Size was 1080 observations over 82 variables. The 77 protein attributes are continuous numerical data, with NaN values present.
`colnames(ds)` | Confirming the column names present in the dataset. The first column is MouseID (12 - 15 observations per mouse, per protein). The final four columns contain the categorical data about each mouse.
`MouseID <- ds$MouseID` | Each mouseID is appended with the suffix '_n' (where 1 <- n <- 15). For example MouseID '309' is recorded as 309_1, 309_2, ... 309_15.

-------------------

### Data Preprocessing - Iteration 1
The native data set will be preprocessed in two iterations for use in our analysis. These data were collected such that they are all on the same scale (approximately 0 - 4) and are intended for direct comparison without any transformation (for example a value of 0.1 for protein A and 1.0 for protein B shows that protein B is being expressed 10 times protein A). Therefore no normalization was needed for these data.

#### Stripping the measurement count from each mouse
During data collection, per each individual mouse, per each protein, 12 - 15 expression measurements were taken _in vitro_. As mentioned in the summary above, these were denoted as '_n' (where 1 <- n <- 15). For the purpose of this investigation, the _n notation is not required, and will be stripped from the data. 
```{r cleaning the data - Stripping MouseID_n, echo=TRUE, tidy=TRUE}
MouseID <- gsub("\\_.*", "", ds$MouseID)
ds$MouseID <- MouseID
```
The function `table(head(ds$MouseID, 150))` samples the first 10 mice in the data set, showing successful removal of _n notation.
```{r checking the data - ds$MouseID cleaned (first 10 mice), echo=TRUE, tidy=TRUE}
table(head(ds$MouseID, 150))
```
<br>

#### Selecting the target proteins & creating a new dataset
Through the use of self-organizing feature maps, Higuera et al were able to determine the most discriminant proteins in comparison of the control mice classes and the trisomic mice classes^1^. For this analysis, these tabulated results were assessed in a semi-quantitative fashion[^8] and the proteins ARC_N, BRAF_N, DYRK1A_N, ITSN1_N, pERK_N, pNUMB_N, S6_N & SOD1_N were selected. A vector of these identifiers was created as `target_attributes_proteins`. These target proteins, along with the categorical tags Genotype, Treatment, Behavior and class were added to another vector, `target_attributes_all`. A new data set was then created, with the native data set observations for only the variables named in the `target_attributes_all` vector.
```{r vector of target proteins for statistical analysis, echo=TRUE, tidy=TRUE}
target_attributes_proteins <- c("ARC_N", "BRAF_N", "DYRK1A_N", "ITSN1_N", "pERK_N",
                           "pNUMB_N", "S6_N", "SOD1_N")
```

```{r vector of all target attributes for statistical analysis, echo=TRUE, tidy=TRUE}
target_attributes_all <- c("MouseID","ARC_N", "BRAF_N", "DYRK1A_N", "ITSN1_N", "pERK_N",
                           "pNUMB_N", "S6_N", "SOD1_N", "Genotype","Treatment","Behavior","class")
```

```{r creating ds_filtered, echo=TRUE, tidy=TRUE}
ds_filtered <- ds[,target_attributes_all]
```

#### Checking the filtered data set
The categorical values in `ds_filtered` were checked with `unique(ds_filtered[,10:13])`, not typographical errors were present, as below.
```{r filtering the categorical values, echo=TRUE, tidy=TRUE}
categorical_counts <- ds_filtered[,10:13]
unique(categorical_counts)
```

The categorical variable counts are summarized in the below (functions in column one).

 Function | Counts | Total
----- | ---------------------------------------- | ----------
`table(ds_filtered$Genotype)`  | Control = 570,  Ts65Dn = 510            | 1080
`table(ds_filtered$Treatment)` | Memantine = 570,   Saline =  510         | 1080
`table(ds_filtered$Behavior)`  | C/S = 525, S/C = 555                      | 1080
`table(ds_filtered$class)`     | c-CS-m = 150,  c-CS-s = 135,  c-SC-m = 150,  c-SC-s = 135,   t-CS-m = 135, t-CS-s = 105, t-SC-m = 135, t-SC-s = 135  | 1080

-------------------

### Data Preprocessing - Iteration 2
#### Locating NaN Values
The coordinate locations for NaN values are displayed below (the result output was transposed for readability). The expression `ds_filtered[c(988, 989, 990),c(1, 10, 11, 12, 13)]` was used to identify the mouse/mice and correspoding categories.
```{r locating NaN, echo=TRUE, tidy=TRUE}
t(which(is.na(ds_filtered), arr.ind=TRUE))
```

```{r show rows, echo=TRUE, tidy=TRUE}
ds_filtered[c(988, 989, 990),c(1, 10, 11, 12, 13)]
```

#### Filling the NaN Values

Through use of `table(ds_filtered$class)` we can see that n = 105 measurements were taken for class 't-SC-s'. As outlined in figure 1, 't-SC-c' corresponds to mice with the Ts65Dn genotype (trisomy, t), not stimulated to learn (Shock Context, SC) and saline treatment (saline, s). The mean value for class 't-SC-s' (n = 105), will be used to fill the NaN values.

```{r t-SC-s filter, echo=TRUE, tidy=TRUE}
tSCs_filter <- ds_filtered$class == "t-SC-s"
NaN_column_mask <- c(3, 4, 5, 6, 7, 9)
NaN_3426_tSCs <- ds_filtered[tSCs_filter, NaN_column_mask]
summary_3426_tSCs <- summary(NaN_3426_tSCs)
summary_3426_tSCs
```

```{r, echo=TRUE, tidy=TRUE}
write.csv(ds_filtered, file = "mouse_data_filtered.csv")
ds_filled_NaN <- gsheet2tbl('https://docs.google.com/spreadsheets/d/1FKFINItxyosgFhwiihpT3MSuMTxR-LaYPNG_XNctwlU/edit?usp=sharing')
```

The filtered mouse data was exported using `write.csv(ds_filtered, file = "mouse_data_filtered.csv")` and added to Google Sheets. The NaN values were filled as follows BRAF_N = 0.2867, DYRK1A_N = 0.3375, ITSN1_N = 0.5491, pERK_N = 0.3577, pNUMB_N = 0.3577 and SOD1_N = 0.7214. The dataset was then imported as `ds_Filled_NaN`. These data were checked for NaN as follows.

```{r checking NaN in ds_filled, echo=TRUE, tidy=TRUE}
NaN_check <- ds_filled_NaN[tSCs_filter, NaN_column_mask]
summary_NaN_check <- summary(NaN_check)
summary_NaN_check
```

Mean Fill Summary      | BRAF_N  | DYRK1A_N | ITSN1_N | pERK_N  | pNUMB_N | SOD1_N 
---------              | --------|  --------| --------| --------| --------| --------
Native Data Mean       |  0.2867 | 0.3375   | 0.5491  | 0.3577  |  0.3577 |  0.7214 
Preprocessed Data Mean |  0.2867 | 0.3375   | 0.5491  | 0.3577  |  0.3577 |  0.7214
Delta Mean             |  0.0000 |  0.0000  |  0.0000 |  0.0000 |  0.0000 |  0.0000

* All mean values remained the same to four decimal places.

-------------------

### Data Exploration through Visualization
#### Scatter Plot Matrix
The preprocessed data will first be visualized with a scatter plot matrix using the `ggpairs` function. The upper section of the plot will be an xy contour, the lower an xy scatter and the diagonal a density plot. A scatter plot matrix is a suitable at this stage as it gives a quantitative overview of these data and provides direction for futher analysis.

* Note scatter plot matrix code will be displayed only once.

```{r scatter_1, echo = TRUE, message = FALSE, results = 'hide', tidy = TRUE}
scatter_matrix <- ggpairs(data = ds_filled_NaN,
        columns = target_attributes_proteins, 
        mapping = aes(colour = class), 
        diag = list(continuous = wrap("densityDiag", alpha=I(0.1)), mapping = ggplot2::aes(fill=class)),
        upper = list(continuous = wrap("density", alpha = I(0.5)), combo = "box"),
        lower = list(continuous = wrap("points", alpha = I(0.4), size = 0.1)))
scatter_matrix_adjusted <- scatter_matrix + theme(panel.spacing=grid::unit(0,"lines"),
                       axis.text = element_text(size = rel(0.5)),
                       strip.text = element_text(face = "bold", size=7),
                       strip.text.x = element_text(margin = margin(.1, 0, .1, 0, "cm")),
                       strip.text.y = element_text(margin = margin(0, .1, 0, .1, "cm")))
scatter_matrix_adjusted + theme(panel.border = element_rect(fill = NA, colour = "grey30", size = 0.2))
```
The resultant plot 'scatter plot matrix A', shows six of the proteins to be correlated in a linear fashion; 

* DYRK1A_N vs BRAF_N
* ITSN1_N vs BRAF_N
* pERK_N vs BRAF_N
* ITSN1_N vs DYRK1A_N
* pERK_N vs DYRK1A_N
* pERK_N vs ITSN1_N

This will provide direction for subsequent statistical analysis.  

Other trends noticable include inversely proportional relationships (e.g., SOD1_N vs BRAF_N), weak correlation (e.g., pNUMB_N vs BRAF_N) and no correlation (e.g., pNUMB_N vs ARC_N). Also of note, in the columns ARC_N, BRAF_N, DYRK1A_N, ITSN1_N and pERK_N, a series of points (amber colour) are seen to lay outside the main clusters in every indivdual scatter plot. Bar charts for two target proteins exhibiting these outliers were plotted (these proteins were randomely chosen.

#### Bar Chart for pERK & BRAF
The resultant plots show these outliers to be very significantly far from the mean of the relavent class, 'c-CS-s'.

```{r, echo=TRUE, tidy=TRUE}
BRAF_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=BRAF_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "BRAF_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=pERK_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "pERK_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=TRUE, tidy=TRUE, fig.height=3, fig.width=7, fig.align="centre"}
grid.arrange(BRAF_N_Hist, pERK_N_Hist, nrow = 2)
```
The most extreme outlier for each protein was determined using the `outlier()` function, inconjunction with filtering.
```{r, echo=TRUE, tidy=TRUE, results = 'hide'}
ds_filled_NaN[ds_filled_NaN$ARC_N == outlier(ds_filled_NaN$ARC_N), c(1,2,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$BRAF_N == outlier(ds_filled_NaN$BRAF_N),c(1,3,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$DYRK1A_N == outlier(ds_filled_NaN$DYRK1A_N),c(1,4,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$ITSN1_N == outlier(ds_filled_NaN$ITSN1_N),c(1,5,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$pERK_N == outlier(ds_filled_NaN$pERK_N),c(1,6,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$pNUMB_N == outlier(ds_filled_NaN$pNUMB_N),c(1,7,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$S6_N == outlier(ds_filled_NaN$S6_N),c(1,8,10,11,12,13)]
ds_filled_NaN[ds_filled_NaN$SOD1_N == outlier(ds_filled_NaN$SOD1_N),c(1,9,10,11,12,13)]
```

Max. Outlier Summary | ARC_N     |  BRAF_N  | DYRK1A_N | ITSN1_N | pERK_N  | pNUMB_N | S6_N   | SOD1_N 
-------              | --------  |  --------| -------- | --------| --------| --------|--------|--------
Value                | 0.0673    | 2.1334   | 2.5164   | 2.6027  |  3.5667 | 0.6311  | 0.8226 | 1.8729  
MouseID              | 3415      | 3484     | 3484     | 3484    |  3484   | 3497    | 3483   | 3411
Genotype             | Control   | Control  | Control  | Control | Control | Control | Ts65Dn | Ts65Dn
Treatment            | Memantine | Saline   | Saline   | Saline  | Saline  | Saline  | Saline | Memantine
Behaviour            | C/S       | C/S      | C/S      | C/S     | C/S     | C/S     | C/S    | S/C
Class                | c-CS-m    |  c-CS-m  | c-CS-m   | c-CS-s  | c-CS-s  | c-CS-s  | t-CS-s | t-SC-m

* Mouse ID 3484 contributed 50% of the maximum outliers, and was dropped from the dataset.

### Data Preprocessing - Iteration 3
#### Dropping MouseID 3484
```{r, echo = TRUE}
ds_filled_NaN <- ds_filled_NaN[!(ds_filled_NaN$MouseID == 3484),]
```

* This gives a resultant dataset of size [13 variables by 1065 observations].

### Data Exploration through Visualization
#### Scatter Plot Matrix - MouseID 3484 dropped
A blah blah was plotted ... Plotted using same code as above
```{r echo = FALSE, message = FALSE, results = 'hide', tidy = TRUE}
scatter_matrix <- ggpairs(data = ds_filled_NaN,
        columns = target_attributes_proteins,
        mapping = aes(colour = class),
        diag = list(continuous = wrap("densityDiag", alpha=I(0.1)), mapping = ggplot2::aes(fill=class)),
        upper = list(continuous = wrap("density", alpha = I(0.5)), combo = "box"),
        lower = list(continuous = wrap("points", alpha = I(0.4), size = 0.1)))
scatter_matrix_adjusted <- scatter_matrix + theme(panel.spacing=grid::unit(0,"lines"),
                       axis.text = element_text(size = rel(0.5)),
                       strip.text = element_text(face = "bold", size=7),
                       strip.text.x = element_text(margin = margin(.1, 0, .1, 0, "cm")),
                       strip.text.y = element_text(margin = margin(0, .1, 0, .1, "cm")))
scatter_matrix_adjusted + theme(panel.border = element_rect(fill = NA, colour = "grey30", size = 0.2))
```

#### Box Plots for each Protein

```{r, echo=FALSE}
ARC_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=ARC_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "ARC_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
BRAF_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=BRAF_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "BRAF_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE, fig.height=3, fig.width=7, fig.align="centre"}
grid.arrange(ARC_N_Hist, BRAF_N_Hist, nrow = 2)
```

```{r, echo=FALSE}
DYRK1A_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=DYRK1A_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "DYRK1A_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
ITSN1_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=ITSN1_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "ITSN1_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE, fig.height=3, fig.width=7, fig.align="centre"}
grid.arrange(DYRK1A_N_Hist, ITSN1_N_Hist, nrow = 2)
```

```{r, echo=FALSE}
pERK_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=pERK_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "pERK_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
pNUMB_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=pNUMB_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "pNUMB_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE, fig.height=3, fig.width=7, fig.align="centre"}
grid.arrange(pERK_N_Hist, pNUMB_N_Hist, nrow = 2)
```

```{r, echo=FALSE}
S6_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=S6_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "S6_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
SOD1_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=SOD1_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank(), axis.title = element_text(size=8)) + 
  labs(title = NULL,  x = "SOD1_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE, fig.height=3, fig.width=7, fig.align="centre"}
grid.arrange(S6_N_Hist, SOD1_N_Hist, nrow = 2)
```

* Discuss hist and also talk about number of point (sample, population, QQ plots ...)

##### Begin box plots (detect and deal with outliers)
##### All 8 classes per 9 target proteins OR ...?

_________________________

### STATS - Linear Relationships

```{r vector for linear analysis, echo=TRUE, tidy=TRUE}
target_attributes_linear <- c("BRAF_N", "DYRK1A_N", "ITSN1_N", "pERK_N")
```

```{r echo = FALSE, message = FALSE, results = 'hide', tidy = TRUE}
scatter_matrix <- ggpairs(data = ds_filled_NaN,
        columns = target_attributes_linear,
        mapping = aes(colour = class),
        diag = list(continuous = wrap("densityDiag", alpha=I(0.1)), mapping = ggplot2::aes(fill=class)),
        upper = list(continuous = wrap("density", alpha = I(0.5)), combo = "box"),
        lower = list(continuous = wrap("points", alpha = I(0.4), size = 0.1)))
scatter_matrix_adjusted <- scatter_matrix + theme(panel.spacing=grid::unit(0,"lines"),
                       axis.text = element_text(size = rel(0.5)),
                       strip.text = element_text(face = "bold", size=7),
                       strip.text.x = element_text(margin = margin(.1, 0, .1, 0, "cm")),
                       strip.text.y = element_text(margin = margin(0, .1, 0, .1, "cm")))
scatter_matrix_adjusted + theme(panel.border = element_rect(fill = NA, colour = "grey30", size = 0.2))
```


SCATTER PLOTS
  
* Based on CLASS
* ITSN1_N vs DYRK1A_N
* pERK_N vs DYRK1A_N
* pERK_N vs BRAF_N
* Enlarged Scatter w/ tansparent hist
* The value is that it is applicable accross all class, genotypes e.t.c, so these proteins aren't amenable to changes in profile accross the experimental conditions described / explored.

## Statistical Analysis Part 1 - Linear Regression and Correlation
ADD GENERIC EXPLANATION
The linear regression model function `lm()` was used, followed by `summary()` to output the required information for interpretation.

* H0:The data do not fit the linear regression model
* HA:The data fit the linear regression model

### ITSN1_N vs DYRK1A_N
#### Fitting the Data to a Linear Regression Model  
* H0: Accross all eight classes of mice, the expression of ITSN1_N was not linearly dependent to the expression of DYRK1A_N
* HA: Accross all eight classes of mice, the expression of ITSN1_N was linearly dependent to the expression of DYRK1A_N

```{r, echo=TRUE, tidy=TRUE}
ITSN1_N_VS_DYRK1A_N <- lm(ITSN1_N ~ DYRK1A_N, data = ds_filled_NaN)
ITSN1_N_VS_DYRK1A_N %>% summary()
```

```{r, echo=TRUE, tidy=TRUE}
ITSN1_N_VS_DYRK1A_N %>% anova()
```
  
* The p-value of the observed F statistic was found to be 2.2e-16. As p is less than the 0.05 level of significance, we reject H0.

#### The Constant, or Intercept Value  

* H0:α=0 
* HA:α≠0

```{r, echo=TRUE, tidy=TRUE}
ITSN1_N_VS_DYRK1A_N %>% summary() %>% coef()
```
  
* t = 32.696, p < 0.001  
The constant is statistically significant at the 0.05 level. This means that there is statistically significant evidence that the constant is not 0. Their would always be some amount of the target proteins being expressed.

```{r, echo=TRUE, tidy=TRUE}
ITSN1_N_VS_DYRK1A_N %>% confint()
```
    
* 95% CI for the Intercept is found to be [0.167, 0.189]  
H0:α=0 is not captured by this interval, so was rejected.

#### The Slope
  
* H0:β=0
* HA:β≠0

```{r, echo=TRUE, tidy=TRUE}
2*pt(q = 82.33567, df = 1063, lower.tail=FALSE)
```

As p < .05, we reject H0. There was statistically significant evidence that OUES was positively related to VO2VO2 max. The 95% CI for b was found to [1.013, 1.062]. This 95% CI does not capture H0, therefore it was rejected. 

There was a statistically significant positive relationship between the expression of DYRK1A_N and ITSN1_N. 

```{r, echo=TRUE, tidy=TRUE, fig.height=3, fig.width=7, fig.align="centre"}
ITSN1_N_VS_DYRK1A_N_XY <- ggplot(data=ds_filled_NaN, aes(x=DYRK1A_N, y=ITSN1_N, colour = class))
ITSN1_N_VS_DYRK1A_N_XY + 
  geom_point(alpha = I(0.9), size = 0.8) + 
  stat_smooth(method = "lm", col = "black", size = 0.5) +
  labs(title = 'ITSN1_N versus DYRK1A_N Protein Expression Scatter Plot',  
       y = 'ITSN1_N', 
       x = 'DYRK1A_N') +
  theme(legend.title = element_blank(), axis.title = element_text(size=8), title = element_text(size=9)) +
  annotate("text", x = 0.75, y = 1.4, label = "R Squared") + 
  annotate("text", x = 0.75, y = 1.3, label = format(summary(lm(ITSN1_N ~ DYRK1A_N, data = ds_filled_NaN))$r.squared, digits = 3))
```


******
* These data exhibited a positive linear trend, when the expression of DYRK1A_N increases, so does that of ITSN1_N.
* This finding tells us that the relationship of expression for ITSN1_N vs DYRK1A_N _NOT DEPENDENT ON THE 8 CLASSES THE MICE WERE GROUPED INTO - VERY INTERESTING_ 

---------------

# pERK_N vs DYRK1A_N

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_DYRK1A_N <- lm(pERK_N ~ DYRK1A_N, data = ds_filled_NaN)
pERK_N_VS_DYRK1A_N %>% summary()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_DYRK1A_N %>% anova()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_DYRK1A_N %>% summary() %>% coef()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_DYRK1A_N %>% confint()
```

```{r, echo=TRUE, tidy=TRUE}
2*pt(q = 83.524345, df = 1063, lower.tail=FALSE)
```

```{r, echo=TRUE, tidy=TRUE, fig.height=3, fig.width=7, fig.align="centre"}
plot(pERK_N ~ DYRK1A_N, data = ds_filled_NaN, xlab = "DYRK1A_N", ylab = "pERK_N")
abline(pERK_N_VS_DYRK1A_N, col = "red")
```



--------

# pERK_N vs BRAF_N

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_BRAF_N <- lm(pERK_N ~ BRAF_N, data = ds_filled_NaN)
pERK_N_VS_BRAF_N %>% summary()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_BRAF_N %>% anova()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_BRAF_N %>% summary() %>% coef()
```

```{r, echo=TRUE, tidy=TRUE}
pERK_N_VS_BRAF_N %>% confint()
```

```{r, echo=TRUE, tidy=TRUE}
2*pt(q = 67.29879, df = 1063, lower.tail=FALSE)
```

```{r, echo=TRUE, tidy=TRUE, fig.height=3, fig.width=7, fig.align="centre"}
plot(pERK_N ~ BRAF_N, data = ds_filled_NaN, xlab = "BRAF_N", ylab = "pERK_N")
abline(pERK_N_VS_BRAF_N, col = "red")
```



_________________________

## Statistical Analysis Part 2 - Two-sample t-tests PAIRED TWO SAMPLE T-TESTS

The bar plots representing ITSN1_N and SOD1_N over all classes of mice were selected from the 'Data Exploration through Visualization' section of the analysis (displayed below). Selection for statistical analysis was performed in a qualitative fashion.  

 Protein      | Qualitative Observation and Proposed Quantitative Analysis
-------- | -----------------------------------------
ITSN1_N       | The distributions and corresponding mean values occur in a series of pairs. It appears that first two components of each class (genotype and learning stimuli) are deterministic in the expression of ITSN1_N. This will be evaluated pair-wise using the two-sample t-test.
SOD1_N | Expression of SOD1_N is determined predominantly by the learning stimuli applied. In the four classes of CS, the expression levels have tight distributions and similar means. In the four classes of SC, the expression levels have wide distributions, with varied means. This will be evaluated ...

```{r, echo=FALSE, fig.height=3, fig.width=7, fig.align="centre"}
grid.arrange(ITSN1_N_Hist, SOD1_N_Hist, nrow = 2)
```



```{r, echo=TRUE, tidy=TRUE}
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_ARC_BRAF = BRAF_N - ARC_N)
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_BRAF_DYRK1A = DYRK1A_N - BRAF_N)
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_DYRK1A_ITSN1 = ITSN1_N - DYRK1A_N)
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_ITSN1_pERK = pERK_N - ITSN1_N)
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_pERK_pNUMB = pNUMB_N - pERK_N)
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_pNUMB_S6 = S6_N - pNUMB_N)
ds_filled_NaN <- ds_filled_NaN %>% mutate(d_S6_SOD1 = SOD1_N - S6_N)
```

```{r, echo=TRUE, tidy=TRUE}
ds_cCSs <- ds_filled_NaN[(ds_filled_NaN$class == 'c-CS-s'),]
```

# ARC & BRAF for cCSs

```{r, echo=TRUE, tidy=TRUE}
t.test(ds_cCSs$d_ARC_BRAF, 
       mu = 0, 
       alternative = "two.sided")
```

```{r, echo=TRUE, tidy=TRUE}
granova.ds(
  data.frame(ds_cCSs$ARC_N, ds_cCSs$BRAF_N),
  xlab = "ARC_N",
  ylab = "BRAF_N"
  )
```

# BRAF & DYRK1A for cCSs

```{r, echo=TRUE, tidy=TRUE}
t.test(ds_cCSs$d_BRAF_DYRK1A, 
       mu = 0, 
       alternative = "two.sided")
```

```{r, echo=TRUE, tidy=TRUE}
granova.ds(
  data.frame(ds_cCSs$BRAF_N, ds_cCSs$DYRK1A_N),
  xlab = "BRAF_N",
  ylab = "DYRK1A_N"
  )
```

# DYRK1A & ITSN1_N for cCSs

```{r, echo=TRUE, tidy=TRUE}
t.test(ds_cCSs$d_DYRK1A_ITSN1, 
       mu = 0, 
       alternative = "two.sided")
```

```{r, echo=TRUE, tidy=TRUE}
granova.ds(
  data.frame(ds_cCSs$DYRK1A_N, ds_cCSs$ITSN1_N),
  xlab = "DYRK1A_N",
  ylab = "ITSN1_N"
  )
```

# ITSN1_N & pERK_N for cCSs

```{r, echo=TRUE, tidy=TRUE}
t.test(ds_cCSs$d_ITSN1_pERK, 
       mu = 0, 
       alternative = "two.sided")
```

```{r, echo=TRUE, tidy=TRUE}
granova.ds(
  data.frame(ds_cCSs$ITSN1_N, ds_cCSs$pERK_N),
  xlab = "ITSN1_N",
  ylab = "pERK_N"
  )
```

# pERK_N & pNUMB for cCSs

```{r, echo=TRUE, tidy=TRUE}
t.test(ds_cCSs$d_pERK_pNUMB, 
       mu = 0, 
       alternative = "two.sided")
```

```{r, echo=TRUE, tidy=TRUE}
granova.ds(
  data.frame(ds_cCSs$ITSN1_N, ds_cCSs$pERK_N),
  xlab = "ITSN1_N",
  ylab = "pERK_N"
  )
```


Then perform linear regression tests


http://www.rpubs.com/smcclatchy/scatterplots
http://www.sthda.com/english/wiki/ggally-r-package-extension-to-ggplot2-for-correlation-matrix-and-survival-plots-r-software-and-data-visualization
https://cran.r-project.org/web/packages/qwraps2/vignettes/summary-statistics.html
http://rmarkdown.rstudio.com/lesson-3.html
http://rmarkdown.rstudio.com/authoring_rcodechunks.html

## Appendix

# References

[^1]: Higuera, C., Gardiner, K. J., & Cios, K. J. (2015). Self-Organizing Feature Maps Identyfy Proteins Critical to Learning in a Mouse Model of Down Syndrome. PLoS ONE, 10(6).
[^2]: Ahmed, M. M., Dhanasekaran, A. R., Block, A., Tong, S., Costa, A. C. S., Stasko, M., & Gardiner, K. J. (2014). Protein Dynamics Associated with Failed and Rescues Learning in the TS65Dn Mouse Model of Down Syndrome. PLoS ONE, 10(3).
[^3]: Costa, A., Scott-McKean, J., & Stasko, M. (2008). Acute injections of the NMDA receptor antagonist memantine rescue performance deficits of the Ts65Dn mouse model of Down syndrome on a fear conditioning test. Neuropsychopharmacology, 33(7), 1624-1632.
[^4]: Davisson, M., Schmidt, C., Reeves, R., Irving, N., Akeson, E., Harris, B., & Bronson, R. (1993). Segmental trisomy as a mouse model for Down syndrome. Prog Clin Biol Res, 384, 117-133.
[^5]: Mitra, A., Blank, M., & Madison, D. (2012). Developmentally altered inhibition in Ts65Dn, a mouse model of Down syndrome. Brain Research, 1440, 1-8.
[^6]: http://journals.plos.org/plosone/article/figure/image?size=medium&id=10.1371/journal.pone.0129126.g001
[^7]: http://archive.ics.uci.edu/ml/ 
[^8]: https://docs.google.com/spreadsheets/d/1scXPvhOh3kmANCAhf1X_CLwJQckOJA-WMP_f9lHVri4/edit?usp=sharing, CONTROL MICE PROTEINS http://www.plosone.org/article/fetchSingleRepresentation.action?uri=info:doi/10.1371/journal.pone.0129126.s003, TRISOMIC MICE PROTEINS, http://www.plosone.org/article/fetchSingleRepresentation.action?uri=info:doi/10.1371/journal.pone.0129126.s004 


Hunter, J. D. (2007). Matplotlib: A 2D graphics environment. Computing in Science & Engineering, 9(3), 90-95.
McKinney, W. (2010). Data Structures for Statistical Computing in Python.

Pedregosa, F., Varoquaux, G., Gramfort, A., Michel, V., Thirion, B., Grisel, O., . . . Duchesnay, É. (2011). Scikit-learn: Machine Learning in Python. Journal of Machine Learning Research, 12, 2825-2830.
Pérez, F., & Granger, B. E. (2007). IPython: A System for Interactive Scientific Computing. Computing in Science & Engineering, 9(3), 21-29. 

R Chunk Options https://yihui.name/knitr/options/
R Density Plot Options http://www.sthda.com/english/wiki/ggplot2-density-plot-quick-start-guide-r-software-and-data-visualization 



```{r, echo=TRUE, tidy=TRUE}
write.csv(ds_filled_NaN, file = "mouse_data_preprocessed.csv")
ITSN1_N_Data <- gsheet2tbl('https://docs.google.com/spreadsheets/d/131cuxxEJ__jexUuEcfjwoDvRJq1FR3XfbP5YAzSD5Bw/edit?usp=sharing')
```
