---
title: "MATH1324 Assignment 4"
author: "Alistair Grevis-James s3644119"
output:
  html_notebook: default
  pdf_document: default
---

# References

## Statistical Evaluation of Mice Protein Expression Data Set

UCI Machine Learning Repository
Center for Machine Learning and Intelligent Systems

#### Abrstract / Executive Summary
The data file ‘Data_Cortex_Nuclear.xls’ (1) was imported (via Google Drive) into the interactive RStudio environment (2). The libraries pandas, matplotlib and numpy were imported as pd, plt and np respectively. Data was then checked for type, sample rows column names and size (found 1080 by 82). Data cleaning was initially performed by selecting for 11 target protein expression attributes, which had previously been found to have high correlations with learning outcomes. The Boolean categorical labels Treatment, Genotype and Behavior were then replaced with binary Boolean values 0 or 1, and the attribute label Class was generated with the resultant binary Boolean labels to produce 8 unique integer Class labels. A scatter matrix of the uncleaned data was checked to provide direction for analysis, followed by removal of outliers and NaN values. Outliers were identified as falling out a 99th percentile range from any raw protein expression data column. During outlier analysis mouse 3484 was identified as having over 60 outlier values, and was removed from the data set. The outliers and NaN values were then filled with a Class mean. 



# Dataset
To explore the differences in the two classification algorithms, we chose to use a dataset freely available from UCI’s machine learning repository (http://archive.ics.uci.edu/ml/). The chosen dataset ‘Mice Protein Expression Data Set’ was generated from experiments by Higuera et al (2015) and Ahmed et al (2014). These projects aimed to understand the impacts of Down Syndrome on learning through analysis of protein expression in mice.
Down Syndrome (DS) has a prevalence globally of 1 in a 1000 live human births, and is the most common genetically defined cause of intellectual disabilities (Costa 2008, Higuera 2015). DS in humans is caused by the presence of an additional chromosome 21, referred to as trisomy (Costa 2008). Protein expression is significantly perturbed by human trisomy 21, leading to the physical and intellectual manifestations associated with DS. Due to its prevalence and health implications, a strong imperative exists to further understand and treat the condition. Davisson et al., successfully manipulated a mouse genome to produce several models of DS in rodents (Davission, 1933). The Ts65Dn mouse is the best-characterized of the DS models (Mitra, 2012). Higuera et al employed Ts65Dn and normal mice in experimental and control groups, exposing them to a range of variables. The rodents were then euthanized and their cortex protein levels were analysed in a quantitative fashion. The resultant data was subject to an unsupervised clustering method known as Self Organising Maps (SOM) followed by labelling, in order to establish causal relationships between learning outcome, substance exposure, genetic structure and protein expression.
Some mice have been stimulated to learn (context-shock) and others have not (shock-context) and in order to assess the effect of the drug memantine in recovering the ability to learn in trisomic mice, some mice have been injected with the drug and others have not. The 8 different classes of mice in the dataset are summarised in Figure 1.


# Data Importation & Exploration

The data analysis toolkit pandas (McKinney 2010), the scientific computing package numpy, the 2D plotting library matplotlib (Hunter 2007) and the library for opening URLs urllib2 (core python module) were imported into IPython (Perez & Granger 2007) as pd, np, plt and urllib2 respectively. The Excel file ‘Data_Cortex_Nuclear.xls’ was imported from UCI’s machine learning repository (http://archive.ics.uci.edu/ml/) IPython and named “allProtein”.

The class .describe was used to check the data was successfully imported in the previous step. The output was as expected, containing the columns MouseID, Protein Expression (77 columns), Genotype, Treatment, Behavior and class, as described in the literature (Higuera 2015). The .describe class also gave us the size of data [1080 rows × 82 columns], which was consistent with the description of “Mice Protein Expression Data Set” (Higuera 2015). Due to the size of the data set some columns and rows were not displayed, so the class .columns was used to check the attributes had been imported as expected. The class .dtypes was then used to check the data was imported as the correct types. The objects Genotype, Treatment, Behavior and class were checked with .value_counts() to look for missing values, typos and assess if these were Boolean, with the results in table 2.

The output above shows no missing values, no typos and shows Genotype, Treatment and Behavior are Boolean. As noted in table 2, the MouseID takes the form tag_n, where n is the number of measurements made. It was noted that the column class contained a lower-case c, which was replaced with an upper-case C and then checked, as below.

The importation and initial exploration demonstrates we have successfully imported our data, of which there is a vast quantity, 1080 rows by 82 columns, and the data is both numerical and categorical. The next step will be selecting a subset of target proteins (based on observations by Higuera et al) for our analysis. NaN values for protein expression will be checked once the subset is chosen, to avoid unnecessary computation.


# Data Cleaning - Selecting the target proteins

Higuera et al observed that of the 77 proteins analysed through self-organizing feature maps, 11 had the greatest correlation with learning outcomes. These were (in no particular order); BRAF_N, pERK_N, S6_N, pGSK3B_N, CaNA_N, CDK5_N, pNUMB_N, DYRK1A_N, ITSN1_N, SOD1_N and GFAP_N. These proteins, along with the tags Genotype, Treatment, Behavior and Class were selected in a new variable, as described below, followed by a brief check of the protein11, showing the size as expected.

As previously mentioned, NaN values for protein expression were to be checked after a subset (protein11) was selected, protein11.describe() was first used to check the numerical data, as seen below. Immediately, we can see the count for BRAF_N, pERK_N, pNUMB_N, DYRK1A_N, ITSN1_N and SOD1_N to be 1077, rather than 1080. 

The following code was used to confirm the NaN values (through one instance, BRAF_N).  


The native data contains both numerical (the protein expression levels) and categorical data (Genotype, Behavior, Treatment and Class). 



Plots for protein expression
https://stackoverflow.com/questions/20074061/ggplot2-multiple-factors-boxplot-with-scale-x-date-axis-in-r

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r, eval=FALSE}
library(rmarkdown)
library(GGally)
library(car)
library(ggplot2)
library(dplyr)
library(gsheet)
library(gridExtra)
library("htmltools")
library(reshape2)
library(granova)
library(psychometric)
library(Hmisc)
library(qwraps2)
url <- 'https://docs.google.com/spreadsheets/d/158zPd4XCYoaXNzHA2OJPkOmzSWYjwFoEuOiMzV1O-rU/edit?usp=sharing'
ds <- gsheet2tbl(url)
```

```{r defining markup language for qwraps2, echo = TRUE}
options(qwraps2_markup = "markdown")
```

### Describe the data set

```{r IMAGES, echo = TRUE}
#![](http://example.com/logo.png)
#![optional caption text](figures/img.png)
```



### Data Exploration Cleaning
These data 
* Checked by summary (missing values noted)
* Checked column names (all good)
* Checked mouseID, all contain _n
* Striped _n
* Checked stripped ds
In the native data set each mouse had measurements 

```{r cleaning the data - Stripping MouseID_n, echo = TRUE}
MouseID <- gsub("\\_.*", "", ds$MouseID)
ds$MouseID <- MouseID
```

* List of desired from literature 
```{r, echo = TRUE}
target_attributes_all <- c("MouseID", "ARC_N", "BRAF_N", "DYRK1A_N", "ITSN1_N", "pERK_N",
                           "pNUMB_N", "S6_N", "SOD1_N", "Genotype","Treatment","Behavior","class")
target_attributes_proteins <- c("ARC_N", "BRAF_N", "DYRK1A_N", "ITSN1_N", "pERK_N",
                           "pNUMB_N", "S6_N", "SOD1_N")
```
### Make list of desired
```{r, echo = TRUE}
ds_filtered <- ds[,target_attributes_all]
```
### Visualization MUST CLEAN FIRST
```{r, echo = TRUE}
ggpairs(ds_filtered[,target_attributes_proteins])
```

Justification - "Because their is such a mix of mice (8 groups) filter for outliers based mean of each group"
Step 1 - Fill NaN (with group mean)
Step 2 - Remove outliers (group mean)

a. Need to find mean of each group
* We need a count here

```{r, echo=TRUE}
categorical_counts <- ds_filtered[,10:13]
unique(categorical_counts)
```
```{r, echo=TRUE}
table(ds_filtered$Genotype)
```
```{r, echo=TRUE}
table(ds_filtered$Treatment)
```
```{r, echo=TRUE}
table(ds_filtered$Behavior)
```

```{r, echo=TRUE}
table(ds_filtered$class)
```

Discuss

Now located the NaN - summarise this with text found to be [], [], [], [], []

```{r locating NaN, echo=FALSE}
which(is.na(ds_filtered), arr.ind=TRUE)
```

```{r show rows, echo=FALSE}
NAN_row_mask <- c(988, 989, 990)
NAN_column_mask <- c(3, 4, 5, 6, 7, 9)
#want to see mouse, and categories
ds_filtered[c(988, 989, 990),c(1, 10, 11, 12, 13)]
```

# Will fill with Class mean
* filling with genotype, treatment or behavior would be make little sense
* Only choice left to fill with mouseID from or class
* Will go with class
* Exploring the intricacies of the different fills is outside the scope of this course

```{r t-SC-s filter, echo=FALSE}
tSCs_filter <- ds_filtered$class == "t-SC-s"
NaN_column_mask <- c(3, 4, 5, 6, 7, 9)
NaN_3426_tSCs <- ds_filtered[tSCs_filter, NaN_column_mask]
summary_3426_tSCs <- summary(NaN_3426_tSCs)
summary_3426_tSCs
```

* Mean summary tricky - just fill 

### Fill with mean...

```{r, echo=FALSE}
write.csv(ds_filtered, file = "mouse_data_filtered.csv")
ds_filled_NaN <- gsheet2tbl('https://docs.google.com/spreadsheets/d/1FKFINItxyosgFhwiihpT3MSuMTxR-LaYPNG_XNctwlU/edit?usp=sharing')
```

* More succinct code?
```{r 3426 filter, echo=FALSE}
NaN_check <- ds_filled_NaN[tSCs_filter, NaN_column_mask]
summary_NaN_check <- summary(NaN_check)
summary_NaN_check
```

```{r locating NaN SUCCINCT, echo=FALSE}
which(is.na(ds_filled_NaN), arr.ind=TRUE)
```

### Begin box plots (detect and deal with outliers)
### All 8 classes per 9 target proteins OR ...?

```{r, echo=FALSE}
native_proteins_summary <- summary(ds_filtered[,target_attributes_proteins])
summary_table(ds_filtered)
```

* Give one dem code then do all plots (add visible code to appendix also)
```{r, echo=FALSE}
ARC_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=ARC_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "ARC_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
BRAF_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=BRAF_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "BRAF_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE}
grid.arrange(ARC_N_Hist, BRAF_N_Hist, nrow = 2)
```

```{r, echo=FALSE}
DYRK1A_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=DYRK1A_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "DYRK1A_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
ITSN1_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=ITSN1_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "ITSN1_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE}
grid.arrange(DYRK1A_N_Hist, ITSN1_N_Hist, nrow = 2)
```

```{r, echo=FALSE}
pERK_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=pERK_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "pERK_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
pNUMB_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=pNUMB_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "pNUMB_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE}
grid.arrange(pERK_N_Hist, pNUMB_N_Hist, nrow = 2)
```

```{r, echo=FALSE}
S6_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=S6_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "S6_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
SOD1_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=SOD1_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "SOD1_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE}
grid.arrange(S6_N_Hist, SOD1_N_Hist, nrow = 2)
```

* Discuss hist and also talk about number of point (sample, population, QQ plots ...)

### Visualization MUST CLEAN FIRST
```{r, echo = FALSE}
scatter_matrix <- ggpairs(data = ds_filled_NaN, # data.frame with variables
        columns = target_attributes_proteins, # columns to plot, default to all.
        #title = "", # title of the plot
        mapping = aes(colour = class), # aesthetics, ggplot2 style
        upper = list(continuous = wrap("density", alpha = 0.5), combo = "box"),
        lower = list(continuous = wrap("points", alpha = 0.4, size = 0.1), 
        combo = wrap("dot", alpha = 0.6, size = 0.2)))
scatter_matrix_adjusted <- scatter_matrix + theme(panel.spacing=grid::unit(0,"lines"),
                       axis.text = element_text(size = rel(0.5)),
                       #axis.text.x = element_text(margin = margin(t = 0.2), vjust = 0.5), 
                       #axis.text.y = element_text(margin = margin(r = 0.2), hjust = 0.5),
                       #axis.title.x = element_text(size = 0.5, angle = 90),
                       strip.text = element_text(face = "bold", size=7),
                       strip.text.x = element_text(margin = margin(.1, 0, .1, 0, "cm")),
                       strip.text.y = element_text(margin = margin(0, .1, 0, .1, "cm")))
scatter_matrix_adjusted + theme(panel.border = element_rect(fill = NA, colour = "grey30", size = 0.2))
```

## MOUSE 3484 = OUTLIER REMOVAL BELOW
```{r, echo = TRUE}
ds_filled_NaN <- ds_filled_NaN[!(ds_filled_NaN$MouseID == 3484),]
```


```{r, echo=FALSE}
ARC_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=ARC_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "ARC_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```
```{r, echo=FALSE}
BRAF_N_Hist <- ggplot(data=ds_filled_NaN, aes(x=ds_filled_NaN$class, y=BRAF_N, fill = ds_filled_NaN$class)) + 
  geom_boxplot() + 
  theme(legend.title=element_blank()) + 
  labs(title = NULL,  x = "BRAF_N Expression", y = NULL) + 
  guides(fill = FALSE) + 
  coord_flip()
```

```{r, echo=FALSE}
grid.arrange(ARC_N_Hist, BRAF_N_Hist, nrow = 2)
```








```{r, echo=FALSE}
native_proteins_summary <- summary(ds_filtered[,target_attributes_proteins])
summary_table(ds_filtered)
```


### Once outliers are removed, choose target protein and to t-tests 

start with DESCRIPTIVE STATS

* Will break down by mice, then compare
two mice from same class
two mice from same treatment
two mice from same genotype
two mice from same behavior
for one target protein

dietGymFood_data_d %>%
  summarise(
  Min = min(d, na.rm = TRUE),
  Q1 = quantile(d, probs = .25, na.rm = TRUE),
  Median = median(d, na.rm = TRUE),
  Q3 = quantile(d, probs = .75, na.rm = TRUE),
  Max = max(d, na.rm = TRUE),
  Mean = mean(d, na.rm = TRUE),
  SD = sd(d, na.rm = TRUE),
  n = n(),
  Missing = sum(is.na(d))
  )
  
Then perform linear regression tests



http://www.rpubs.com/smcclatchy/scatterplots
http://www.sthda.com/english/wiki/ggally-r-package-extension-to-ggplot2-for-correlation-matrix-and-survival-plots-r-software-and-data-visualization
https://cran.r-project.org/web/packages/qwraps2/vignettes/summary-statistics.html
http://rmarkdown.rstudio.com/lesson-3.html
http://rmarkdown.rstudio.com/authoring_rcodechunks.html




mega_iris = data.frame(gg1$all, Species=rep(iris$Species, length=nrow(gg1$all)))
ggplot(mega_iris, aes_string(x = "x", y = "y")) + 
  facet_grid(xvar ~ yvar, scales = "free") + 
  geom_point(aes(colour=Species), na.rm = TRUE, alpha=0.8) + 
  stat_density(aes(x = x, y = ..scaled.. * diff(range(x)) + min(x)), 
               data = gg1$densities, position = "identity", 
               colour = "grey20", geom = "line")

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.



## Appendix

### Data Exploration Cleaning
These data 
```{r checking the data - summary, echo = TRUE}
summary(ds)
```

```{r checking the data - column names, echo = TRUE}
column_names <- colnames(ds)
column_names
```

```{r checking the data - MouseID, echo = TRUE}
MouseID <- ds$MouseID
MouseID
```

```{r checking the data - ds$MouseID cleaned, echo = TRUE}
table(ds$MouseID)
```

```{r, echo = TRUE}
all_target_attributes <- c("MouseID","BRAF_N","pERK_N","S6_N","pGSK3B_N","CaNA_N",
                 "CDK5_N","pNUMB_N","DYRK1A_N","ITSN1_N","SOD1_N","GFAP_N",
                 "Genotype","Treatment","Behavior","class")
expression_attributes <- c("BRAF_N","pERK_N","S6_N","pGSK3B_N","CaNA_N",
                 "CDK5_N","pNUMB_N","DYRK1A_N","ITSN1_N","SOD1_N","GFAP_N")
```



```{r cchecking the data - ds$MouseID cleaned, echo = TRUE}
library(ggplot2)
ggpairs(with(ds, data.frame(MouseID, ITSN1_N)))
```



```{r REPLACE couldnt make work, echo=FALSE}
#df <- data.frame(ds_filtered[c(988, 989, 990), "BRAF_N"] == 0.2867, 
ds_filtered <- ds_filtered[c(988, 989, 990),"BRAF_N"] == 0.2867
dataset <- matrix(sample(c(NA, 1:5), 25, replace = TRUE), 5);
data <- as.data.frame(dataset)
```



our_summary <- 
  with(categorical_counts, 
       list("Genotype" = table(categorical_counts[,1]),
            "Treatment" = table(ds_filtered$Treatment),
            "Behavior" = table(ds_filtered$Behavior),
            "class" = table(ds_filtered$class)
            ))
our_summary



theme_gray
function (base_size = 11, base_family = "") 
{
 half_line <- base_size/2
theme(
  line = element_line(colour = "black", size = 0.5, 
                      linetype = 1, lineend = "butt"), 
  rect = element_rect(fill = "white", colour = "black",
                      size = 0.5, linetype = 1),
  text = element_text(family = base_family, face = "plain",
                      colour = "black", size = base_size,
                      lineheight = 0.9,  hjust = 0.5,
                      vjust = 0.5, angle = 0, 
                      margin = margin(), debug = FALSE), 
  
  axis.line = element_blank(), 
  axis.text = element_text(size = rel(0.8), colour = "grey30"),
  axis.text.x = element_text(margin = margin(t = 0.8*half_line/2), 
                             vjust = 1), 
  axis.text.y = element_text(margin = margin(r = 0.8*half_line/2),
                             hjust = 1),
  axis.ticks = element_line(colour = "grey20"), 
  axis.ticks.length = unit(half_line/2, "pt"), 
  axis.title.x = element_text(margin = margin(t = 0.8 * half_line,
                                          b = 0.8 * half_line/2)),
  axis.title.y = element_text(angle = 90, 
                              margin = margin(r = 0.8 * half_line,
                                          l = 0.8 * half_line/2)),
  
  legend.background = element_rect(colour = NA), 
  legend.margin = unit(0.2, "cm"), 
  legend.key = element_rect(fill = "grey95", colour = "white"),
  legend.key.size = unit(1.2, "lines"), 
  legend.key.height = NULL,
  legend.key.width = NULL, 
  legend.text = element_text(size = rel(0.8)),
  legend.text.align = NULL,
  legend.title = element_text(hjust = 0), 
  legend.title.align = NULL, 
  legend.position = "right", 
  legend.direction = NULL,
  legend.justification = "center", 
  legend.box = NULL, 
  
  panel.background = element_rect(fill = "grey92", colour = NA),
  panel.border = element_blank(), 
  panel.grid.major = element_line(colour = "white"), 
  panel.grid.minor = element_line(colour = "white", size = 0.25), 
  panel.margin = unit(half_line, "pt"), panel.margin.x = NULL, 
  panel.margin.y = NULL, panel.ontop = FALSE, 
  
  strip.background = element_rect(fill = "grey85", colour = NA),
  strip.text = element_text(colour = "grey10", size = rel(0.8)),
  strip.text.x = element_text(margin = margin(t = half_line,
                                              b = half_line)), 
  strip.text.y = element_text(angle = -90, 
                              margin = margin(l = half_line, 
                                              r = half_line)),
  strip.switch.pad.grid = unit(0.1, "cm"),
  strip.switch.pad.wrap = unit(0.1, "cm"), 
  
  plot.background = element_rect(colour = "white"), 
  plot.title = element_text(size = rel(1.2), 
                            margin = margin(b = half_line * 1.2)),
  plot.margin = margin(half_line, half_line, half_line, half_line),
  complete = TRUE)
}

+ theme_linedraw()

axis.text = element_text(size = rel(0.6), colour = "grey30")


```{r, echo = TRUE}
scatter_matrix <- ggpairs(data = ds_filled_NaN, # data.frame with variables
        columns = target_attributes_proteins, # columns to plot, default to all.
        #title = "", # title of the plot
        mapping = aes(colour = class), # aesthetics, ggplot2 style
        upper = list(continuous = wrap("density", alpha = 0.5), combo = "box"),
        lower = list(continuous = wrap("points", alpha = 0.3, size = 0.1), 
                     combo = wrap("dot", alpha = 0.4, size=0.2)))
scatter_matrix + theme(panel.spacing=grid::unit(0,"lines"),
                       axis.text = element_text(size = rel(0.5)),
                       #axis.text.x = element_text(margin = margin(t = 0.2), vjust = 0.5), 
                       #axis.text.y = element_text(margin = margin(r = 0.2), hjust = 0.5),
                       #axis.title.x = element_text(size = 0.5, angle = 90),
                       strip.text=element_text(size=6))
```







